devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -0,0 +1,142 @@
+/**
+ *  Copyright 2018 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (name: "ZigBee Accessory Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.remotecontroller", runLocally: false, minHubCoreVersion: '000.019.00012', executeCommandsLocally: false, mnmn: "SmartThings", vid: "generic-dimmer") {
+		capability "Actuator"
+		capability "Configuration"
+		capability "Switch"
+		capability "Button"
+		capability "Switch Level"
+
+		fingerprint profileId: "0104", inClusters: "0000,0003", outClusters: "0000,0004,0003,0006,0008,0005", manufacturer: "Aurora", model: "Remote50AU", deviceJoinName: "Aurora Wireless Wall Remote"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getSTEP() {10}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap && descMap.clusterInt == 0x0006) {
+			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+				if (device.currentValue("level") == 0) {
+					sendEvent(name: "level", value: STEP)
+				}
+				sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
```

Comment:
This device has explicit On and Off buttons, correct? If so I think it may be better to create the `on` event on receipt of the `On` command rather than toggling. Should also force each event to be a state change event.

Reply:
Only problem is that then we have to maintain the state and there's no way to get that from the device. (and it's not indicated on the device anywhere)

Reply:
a single button

Reply:
As I commented above, we can use the command it sends us to know.  If the command ID is `0x01` it's an on and `0x00` is an off.  I think we could consider the argument that we should explicitly mark the event with `isStateChange: true` as well, but I don't feel too strongly on that.